### ISSUE TITLE
ESRI styles with text seem to crash rendering

### ISSUE THREAD

**Issue by debaser75** [openlayers/ol-mapbox-style#244]
I'm hoping to render ESRI vector tiles with and ESRI style. All works well when I add a style that has no fonts. But if I attempt to add a style that has fonts, I get an error deep in the OL source code and the map doesn't render.

```
<script src="https://cdnjs.cloudflare.com/ajax/libs/openlayers/4.3.0/ol.js"></script>
<script src = "https://cdn.jsdelivr.net/npm/ol-mapbox-style@4.3.0/dist/olms.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/openlayers/4.3.0/ol.css" type="text/css">

       var vectorTiles = new ol.layer.VectorTile({
      source: new ol.source.VectorTile({
        format: new ol.format.MVT(),
        
        tileGrid: ol.tilegrid.createXYZ({maxZoom: 22}),
        tilePixelRatio: 16,
        name:'vector',
        url: 'https://basemaps.arcgis.com/arcgis/rest/services/World_Basemap_v2/VectorTileServer/tile/{z}/{y}/{x}.pbf',
      })
    })


 $.get("https://www.arcgis.com/sharing/rest/content/items/e7c0e85210bf406e9c329a0030f67fbc/resources/styles/root.json")
            .then(function(res) {
            olms.applyStyle(vectorTiles, res, "esri");
 }); 
```

That style is a derivative of the "Gray Base". It's only the base layers, no text (ESRI provides another gray base map that has all of the text).  But if I try the following topo style, the map doesn't render:

I looked up all of the fonts in the style and made sure I put them in the HTML header (Noto Sans, Noto Serif, Roboto Condensed).

Any idea what's happening? What can I do to fix this?

Thanks!

**Comment by ahocevar** [comment]
As a first step, please use the latest versions of ol and ol-mapbox-style. In the issue description above, you forgot to paste the failing style. But I think you meant ESRI's World Topographic Map (v2). With the latest version of ol-mapbox-style, your code becomes much simpler:
```js
import apply from 'ol-mapbox-style';

apply('map', 'https://www.arcgis.com/sharing/rest/content/items/7dc6cea0b1764a1f9af2e679f642f0f5/resources/styles/root.json');

```
See https://codesandbox.io/s/vector-tiles-4326-k3o6x for the above code in action.

**Comment by debaser75** [comment]
Thanks for the quick response! And thank you for providing this very valuable tool.

I should have explained: I tried upgrading my web maps to OL 5.1 a while back, but Safari users were experiencing a lot of rendering problems. So I gave up took it back to OL 4.3, especially since I don't own any Apple devices and wanted to stop bugging my friends with "Is it working now?". V3.5 of ol-mapbox-style was in one of the few examples I could find that doesn't use 'import' - I'm very new to that approach in JS/HTML and I'm hoping to not have to run Node or something similar (at least on the server, I don't mind having it on my dev computer)
